### PR TITLE
✨ feat: download request persistence

### PR DESCRIPTION
**Pull request by heathharrelson**
Persist information collected during the download request workflow, including:

* user details
* research purpose, license agreement, and attestation
* selected folders

### CODE REVIEW

File: pages/datasets/[datasetid]/access/license.vue [hunk 1]
```diff
@@ -19,6 +22,11 @@ if (error.value) {
   throw new Error("Failed to fetch dataset");
 }
 
+if (researchPurpose.value === "") {
+  console.warn("Redirecting to collect research purpose");
```
Comment (kirenotneb) [inline pages/datasets/[datasetid]/access/license.vue:26]: Do you want to keep this warn?

**Review by kirenotneb — APPROVED** [review]
Looks good!

File: pages/datasets/[datasetid]/access/license.vue (hunk 1 above)
Comment (heathharrelson) [reply pages/datasets/[datasetid]/access/license.vue:26]: Yes, this one is intentional.

File: pages/datasets/[datasetid]/access/select.vue [hunk 2]
```diff
@@ -7,6 +7,8 @@ const route = useRoute();
 
 const { datasetid } = route.params as { datasetid: string };
 const { data: dataset, error } = await useDataset(datasetid);
+const { data: agreement } = await useDownloadAgreement(datasetid);
+console.log(agreement.value);
```
[the remark this answers is not in the document]
Comment (heathharrelson) [reply pages/datasets/[datasetid]/access/select.vue:11]: Thanks,  I missed this one when cleaning up.

**Pull request merged**